### ISSUE TITLE
"Export Schema To Data Files" fails

### ISSUE THREAD

**Issue by philipdanielhayton** [vendrhub/vendr-deploy#17]
Hi Matt,

As per your comment on the forum I've had a play with Vendr.Deploy 3.0.0-rc00012, the good news is the project starts without any issue, the bad news is that the deploy export doesnt work, it's throwing the following:

    {
      "Date": "2022-09-27T21:04:13.27913+01:00",
      "Message": "Unknown entity type \"vendr-store\".",
      "Ex": {
        "ClassName": "System.ArgumentException",
        "Message": "Unknown entity type \"vendr-store\".",
        "Data": null,
        "InnerException": null,
        "HelpURL": null,
        "StackTraceString": "   at Umbraco.Cms.Core.UdiParser.<>c__DisplayClass13_0.<GetRootUdi>b__0(String x)\r\n   at 
    System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)\r\n   at 
    Umbraco.Deploy.Infrastructure.Work.WorkItems.DiskDumpWorkItem.<>c.<ExecuteAsync>b__12_0(String x)\r\n   at 
    System.Linq.Enumerable.SelectListIterator`2.ToList()\r\n   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)\r\n   at 
    Umbraco.Deploy.Infrastructure.Work.WorkItems.DiskDumpWorkItem.ExecuteAsync(IWorkContext context, CancellationToken token)",
        "RemoteStackTraceString": null,
        "RemoteStackIndex": 0,
        "ExceptionMethod": null,
        "HResult": -2147024809,
        "Source": "Umbraco.Core",
        "WatsonBuckets": null,
        "ParamName": null
      },
      "Sender": "Umbraco.Deploy.Infrastructure.Work.WorkItems.DiskDumpWorkItem"
    }

**Steps to reproduce**

1. On a project with VendrDeploy installed...
2. Go to Settings > Deploy
3. Click "Export Schema To Data Files"

**Additional context**

Just to confirm that this is an issue with Vendr.Deploy I uninstalled / reinstalled it several times, the export works without issue when Vendr.Deploy is disabled, but is obviously missing the Vendr entities.

**Version info**

Umbraco 10.2.1
Vendr 3.0.1-beta0054
Vendr.Deploy 3.0.0-rc00012

**Comment by mattbrailsford** [comment]
Hmmm, that is weird. 

We have a composer / component that registers the entity type.

Is there any other error in the trace log that might be linked to this? Maybe there is something erroring in the composer / component that causes that registration to fail.

**Comment by mattbrailsford** [comment]
I'm going to setup a v10 Umbraco Cloud site today so I can do more thurough testing on this

**Comment by philipdanielhayton** [comment]
> Hmmm, that is weird.
> 
> We have a composer / component that registers the entity type.
> 
> Is there any other error in the trace log that might be linked to this? Maybe there is something erroring in the composer / component that causes that registration to fail.

No, when I restart the site the only error I'm getting is from the failed deployment task, which looks like it automatically runs when the site is started.

Edit: Sorry, it's not running automatically on startup, that's just me being a derp.

**Comment by mattbrailsford** [comment]
No worries,

I got this replicating locally and spoke to Andy at HQ and apparantley now we need to register our custom UDI types at startup which we didn't previously need to do.

Without the UDI types saving an entity would create the schema on disk for that entity, but for the "Export Schemas" button to work it needs to  know the UDI type and so I've now registered these and it's all working.

I'm just currently working on getting product attributes / presets being able to transfer so will push another update to the unstable feed shortly.
